service/src/test/java/greencity/service/DiscountServiceImplTest.java

```diff
@@ -0,0 +1,87 @@
+package greencity.service;
+
+import greencity.dto.discount.DiscountValueVO;
+import greencity.entity.DiscountValue;
+import greencity.exception.exceptions.NotFoundException;
+import greencity.repository.DiscountValuesRepo;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.TypeToken;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class DiscountServiceImplTest {
+    private final DiscountValueVO discountValueVO = DiscountValueVO.builder()
+        .id(1L)
+        .value(10)
+        .build();
+    private final DiscountValue discountValue = DiscountValue.builder()
+        .id(1L)
+        .value(10)
+        .build();
```

Comment:
Better move such models to some Utils class. Also, better use Random class for digits